[CIS-1002] Message List Documentation (#1437)
GetStream/stream-chat-swift: docusaurus/docs/iOS/components/thread.md

    ComposerVC [href="../message-composer"]
` }</Digraph>

### Overview

- [`ChatThreadHeaderView`](../thread-header-view) is responsible to display the thread information in the `navigationItem.titleView`.
- [`ChatMessageListVC`](../message-list) is the component that handles the rendering of the replies.
- [`ComposerVC`](../message-composer) is the component that handles the creation of new replies.

## Usage
By default, the `ChatThreadVC` is created when you click to see the replies of a message in the `ChatMessageListVC` component. But in case you want to create it programmatically, you can use the following code:

```swift
let cid = "channel-id"
let messageId = "message-id"
let threadVC = ChatThreadVC()
threadVC.channelController = ChatClient.shared.channelController(for: cid)
threadVC.messageController = ChatClient.shared.messageController(
    cid: cid,
    messageId: messageId
)

navigationController?.show(threadVC)
```

<SingletonNote />

## UI Customization

You can customize how the `ChatThreadVC` looks by subclassing it and swap the component in `Components` config:

```swift
Components.default.threadVC = CustomChatThreadVC.self
```

<ComponentsNote />

> Just like the `ChatChannelVC`, the `ChatThreadVC` is only responsible for composing the `ChatThreadHeaderView`, `ChatMessageListVC` and `ChatMessageComposerVC` components together. In case you want to customize the rendering of the replies, you should read the [Message List](../message-list) documentation and the [Message](../message) documentation.

Reviewer: ```suggestion
Just like the `ChatChannelVC`, the `ChatThreadVC` is only responsible for composing the `ChatThreadHeaderView`, `ChatMessageListVC` and `ChatMessageComposerVC` components together. In case you want to customize the rendering of the replies, have a look at [Message List](../message-list) documentation and the [Message](../message) documentation.
```
